extensions/azurecore/src/account-provider/azureAccountProvider2.ts

```diff
@@ -0,0 +1,443 @@
+	}
+
+	private async _getSecurityToken(account: azdata.Account, resource: azdata.AzureResource): Promise<{}> {
+		return this.getAccessTokens(account, resource);
+	}
+
+	// interface method
+	prompt(): Thenable<azdata.Account | azdata.PromptFailedResult> {
+		return this._prompt();
+	}
+
+
+	private async _prompt(): Promise<azdata.Account | azdata.PromptFailedResult> {
+		if (this.isInitialized === false) {
+			vscode.window.showInformationMessage(notInitalizedMessage);
+			return { canceled: false };
+		}
+		const pathMappings = new Map<string, (req: http.IncomingMessage, res: http.ServerResponse, reqUrl: url.UrlWithParsedQuery) => void>();
+
+		// const redirectUri = await vscode.env.createAppUri({
+		// 	payload: { path: 'authenticated' }
+		// });
+		const nonce = crypto.randomBytes(16).toString('base64');
+
+		const server = this.createAuthServer(pathMappings);
+
+		const port = await this.listenToServer(server);
+		try {
+			const authUrl = this.createAuthUrl(
+				this.metadata.settings.host,
+				AzureAccountProvider.redirectUrlAAD,
+				this.metadata.settings.clientId,
+				this.metadata.settings.signInResourceId,
+				AzureAccountProvider.AadCommonTenant,
+				`${port},${encodeURIComponent(nonce)}`
+			);
+
+			this.addServerPaths(pathMappings, nonce, authUrl);
+
+			const accountAuthenticatedPromise = new Promise<AzureAccount>((resolve, reject) => {
+				AzureAccountProvider.eventEmitter.on(AzureAccountProvider.AzureAccountAuthenticatedEvent, ({ account, error }) => {
+					if (error) {
+						return reject(error);
+					}
+					return resolve(account);
+				});
+			});
+
+			const urlToOpen = `http://localhost:${port}/signin?nonce=${encodeURIComponent(nonce)}`;
+
+			vscode.env.openExternal(vscode.Uri.parse(urlToOpen));
+
+			const account = await accountAuthenticatedPromise;
+
+			return account;
+		} finally {
+			server.close();
+		}
+	}
+
+	private addServerPaths(
+		pathMappings: Map<string, (req: http.IncomingMessage, res: http.ServerResponse, reqUrl: url.UrlWithParsedQuery) => void>,
+		nonce: string,
+		authUrl: string) {
+
+		const initialSignIn = ((req: http.IncomingMessage, res: http.ServerResponse, reqUrl: url.UrlWithParsedQuery) => {
+			const receivedNonce = (reqUrl.query.nonce as string || '').replace(/ /g, '+');
+			if (receivedNonce !== nonce) {
+				// TODO handle broken nonce situation
+				return;
+			}
+			res.writeHead(302, { Location: authUrl });
+			res.end();
+		});
+
+		const callback = ((req: http.IncomingMessage, res: http.ServerResponse, reqUrl: url.UrlWithParsedQuery) => {
+			const state = reqUrl.query.state as string ?? '';
+			const code = reqUrl.query.code as string ?? '';
+
+			const stateSplit = state.split(',');
+			if (stateSplit.length !== 2) {
+				// TODO handle broken state.
+				return;
+			}
+
+			if (stateSplit[1] !== nonce) {
+				// TODO handle broken nonce situation
+				return;
+			}
+
+			res.writeHead(200, 'success, you may close this page');
```

Comment:
Are you still working on this?

Reply:
Yeah eventually we might want to show a webpage that signin was successful. It's fine for a crawl phase I assume.

Reply:
Let me know if we should have something prettier here to get this merged in :)